### PR TITLE
Feature/379 automated performance tests

### LINKED ISSUE #379: Research: Research a way to perform automated performance tests
**Is your feature request related to a problem? Please describe.**
As you can see in this issue: https://github.com/DigitalExcellence/dex-backend/issues/376. We found out that the current use of getting projects can be optimized to be faster. Other developers however found that the current use of getting projects is faster. Due to this difference it is hard to know the exact performance of the application.

**Describe the solution you'd like**
Research a way to automate performance tests for production (and other) servers.

### PR DESCRIPTION
**Pull request by BluRRayS**
## Description

Executing performance tests against staging or production with K6.
I, Ramon And Tim have written some tests to test the performance of the getProjects endpoint since it is a hot endpoint and often used. We wrote Load, Soak and stress tests and we wrote documentation on how to write and execute tests in the Wiki.

The documentation can be found [here](https://github.com/DigitalExcellence/dex-backend/wiki/Perfomancing-testing).
And the research [here](https://docs.google.com/document/d/1zfNJ88HpEvLdgeSNNsJiQRV1w5-vwHYIBbZMNggbR7U/edit).

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [X] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [X] My code follows the style guidelines of this project
-   [X] I did not update API Controllers, if I did, I added/changed Postman tests
-   [X] I have performed a self-review of my own code
-   [X] I have commented my code, particularly in hard-to-understand areas
-   [X] I have made corresponding changes to the documentation
-   [X] I have added tests that prove my fix is effective or that my feature works
-   [X] New and existing unit tests pass locally with my changes
-   [X] I updated the changelog with an end-user readable description
-   [X] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

You can run the performance tests using any CLI on a system that has docker and docker-compose installed.
Run tests by navigating to the `PerformanceTests` folder and then executing the commands state in the documentation found [here](https://github.com/DigitalExcellence/dex-backend/wiki/Perfomancing-testing)

[#379 ](https://github.com/DigitalExcellence/dex-backend/issues/379)

Closes #379 Performance tests

### CODE REVIEW

**Review by niraymak — APPROVED** [review]
Great job guys! Good start for performance testing! 🥇

File: API/appsettingsapi.Development.json [hunk 1]
```diff
@@ -37,8 +37,8 @@
         },
         "RabbitMQ": {
             "Hostname": "localhost",
-            "Username": "test",
-            "Password": "test"
+            "Username": "guest",
+            "Password": "guest"
```
Comment (Maxvanhattum) [inline API/appsettingsapi.Development.json:41]: If using docker leave it on test test

Comment (niraymak) [inline API/appsettingsapi.Development.json:41]:
```suggestion
            "Username": "test",
            "Password": "test"
```

**Review by niraymak — CHANGES_REQUESTED** [review]

**Review by Maxvanhattum — APPROVED** [review]
Seems to be a good basis

**Review by niraymak — APPROVED** [review]

**Review by Maxvanhattum — APPROVED** [review]

**Pull request merged**